apps/wallet-mobile/src/features/Notifications/useCases/common/SwipeOutWrapper.tsx

```diff
@@ -1,18 +1,42 @@
 import * as React from 'react'
-import {Animated, Dimensions, PanResponder} from 'react-native'
+import {Animated, Dimensions, Easing, PanResponder} from 'react-native'
 
 type Props = {
   children: React.ReactNode
   onSwipeOut: () => void
+  onExpired: () => void
 }
 
-export const SwipeOutWrapper = ({children, onSwipeOut}: Props) => {
-  const {pan, panResponder} = usePanAnimation({onRelease: onSwipeOut})
+const notificationDisplayTime = 20 * 1000 // 20 seconds
+const fadeInTime = 200
+const fadeOutPaddingTime = 100
+
+export const SwipeOutWrapper = ({children, onSwipeOut, onExpired}: Props) => {
+  const {pan, panResponder, fadeIn, opacity, fadeOut, translateY} = usePanAnimation({onRelease: onSwipeOut})
+  const onExpiredRef = React.useRef(onExpired)
+  onExpiredRef.current = onExpired
+
+  React.useEffect(() => {
+    const expiredTimeout = setTimeout(() => onExpiredRef.current(), notificationDisplayTime)
+    const fadeOutTimeout = setTimeout(() => fadeOut(), notificationDisplayTime - fadeInTime - fadeOutPaddingTime)
+
+    return () => {
+      clearTimeout(expiredTimeout)
+      clearTimeout(fadeOutTimeout)
+    }
+  }, [fadeIn, fadeOut])
+
+  React.useEffect(() => {
+    // When executed without setTimeout, the animation does not start
```

Comment:
2 guesses here, probably mounting issue, try to wrap the fadeIn within the `useLayoutEffect` it should work, otherwise you can leverage -> `requestAnimationFrame` as well.